<a href="https://colab.research.google.com/github/JSEFERINO/Estadistica_Moderna/blob/main/09_C_Datos_Est_Descriptiva.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **Curso Introductorio a la Ciencia de Datos por Julio Hurtado Márquez**
## **Parte II**
# **42. Importar y exportar dataframes limpios**

In [ ]:
import pandas as pd
import numpy as np

In [ ]:
# **b. Construimos el DataFrame con el archivo excel**
df_datos2023 = pd.read_csv("/content/df_datos_LIMPIOS.csv", delimiter=',')
df_datos2023

CURSO  DEFINITIVA PROGRAMA  EDAD URBANO  PESO  ESTATURA  \
0    ESTADISTICA I         4.3     COMS    18     SI    55       164   
1    ESTADISTICA I         3.1     COMS    19     SI    65       165   
2    ESTADISTICA I         4.2     DERH    25     NO    56       170   
3    ESTADISTICA I         3.6     DERH    23     SI    65       170   
4    ESTADISTICA I         3.9     PSIC    19     SI    50       169   
..             ...         ...      ...   ...    ...   ...       ...   
116  ESTADISTICA I         3.5     PSIC    18     SI    60       163   
117  ESTADISTICA I         3.5     PSIC    18     SI    60       180   
118  ESTADISTICA I         2.5     COMS    21     NO    45       172   
119  ESTADISTICA I         3.0     COMS    20     NO    50       160   
120  ESTADISTICA I         3.5     DERH    21     NO    57       160   

     SATISFACC  SABER 11  PROMEDIO SEXO ESTRATO Herramientas  
0            4       350      4.26    F      II         PTWF  
1            5       328      4.11    F       I         CYDT  
2            5       285      3.70    M       I         CYDT  
3            5       280      4.10    F      II         PCWF  
4            5       319      4.00    F      II         CYWF  
..         ...       ...       ...  ...     ...          ...  
116          5       310      3.90    F     III         PTWF  
117          3       346      3.90    M      IV         PCWF  
118          5       221      3.70    F      II         CYWF  
119          4       332      3.50    F      II         CYDT  
120          4       293      3.50    F     III         PTWF  

[121 rows x 13 columns]

# **43. Vamos agregar una nueva columna a nuestra base de datos**

## **a. Se hace necesario importar la libreria _numpy_ en caso de no haberla escrito arriba**

In [ ]:
## **a. Se hace necesario importar la libreria _numpy_ en caso de no haberla escrito arriba**
import numpy as np

## **b. Vamoa crear unn _array_ de 121 elemetos usando la libreria _numpy_ y el metodo _arange_**
- Nos crea una lista de números desde 0 al 120

In [ ]:
## **b. Vamoa crear unn _array_ de 121 elemetos usando la libreria _numpy_ y el metodo _arange_**
np.arange(0,121)

array([  0,   1,   2,   3,   4,   5,   6,   7,   8,   9,  10,  11,  12,
        13,  14,  15,  16,  17,  18,  19,  20,  21,  22,  23,  24,  25,
        26,  27,  28,  29,  30,  31,  32,  33,  34,  35,  36,  37,  38,
        39,  40,  41,  42,  43,  44,  45,  46,  47,  48,  49,  50,  51,
        52,  53,  54,  55,  56,  57,  58,  59,  60,  61,  62,  63,  64,
        65,  66,  67,  68,  69,  70,  71,  72,  73,  74,  75,  76,  77,
        78,  79,  80,  81,  82,  83,  84,  85,  86,  87,  88,  89,  90,
        91,  92,  93,  94,  95,  96,  97,  98,  99, 100, 101, 102, 103,
       104, 105, 106, 107, 108, 109, 110, 111, 112, 113, 114, 115, 116,
       117, 118, 119, 120])

## **c. Verifiquemos su longitud**

In [ ]:
## **c. Verifiquemos su longitud**
len(np.arange(0,121))

121

## **d. Coloquemosle el nombre a esta nueva columna**

In [ ]:
## **d. Coloquemosle el nombre a esta nueva columna**
Aris2 = np.arange(0,121)


## **e. Hagamos una copia de nuestra base de datos**

In [ ]:
## **e. Hagamos una copia de nuestra base de datos**
df_datos2 = df_datos2023.copy()
df_datos2

CURSO  DEFINITIVA PROGRAMA  EDAD URBANO  PESO  ESTATURA  \
0    ESTADISTICA I         4.3     COMS    18     SI    55       164   
1    ESTADISTICA I         3.1     COMS    19     SI    65       165   
2    ESTADISTICA I         4.2     DERH    25     NO    56       170   
3    ESTADISTICA I         3.6     DERH    23     SI    65       170   
4    ESTADISTICA I         3.9     PSIC    19     SI    50       169   
..             ...         ...      ...   ...    ...   ...       ...   
116  ESTADISTICA I         3.5     PSIC    18     SI    60       163   
117  ESTADISTICA I         3.5     PSIC    18     SI    60       180   
118  ESTADISTICA I         2.5     COMS    21     NO    45       172   
119  ESTADISTICA I         3.0     COMS    20     NO    50       160   
120  ESTADISTICA I         3.5     DERH    21     NO    57       160   

     SATISFACC  SABER 11  PROMEDIO SEXO ESTRATO Herramientas  
0            4       350      4.26    F      II         PTWF  
1            5       328      4.11    F       I         CYDT  
2            5       285      3.70    M       I         CYDT  
3            5       280      4.10    F      II         PCWF  
4            5       319      4.00    F      II         CYWF  
..         ...       ...       ...  ...     ...          ...  
116          5       310      3.90    F     III         PTWF  
117          3       346      3.90    M      IV         PCWF  
118          5       221      3.70    F      II         CYWF  
119          4       332      3.50    F      II         CYDT  
120          4       293      3.50    F     III         PTWF  

[121 rows x 13 columns]

## **f. Agreguemos ahora a nuestro nuevo DataFrame la columna que hemos creado**

In [ ]:
## **f. Agreguemos ahora a nuestro nuevo DataFrame la columna que hemos creado**
df_datos2["Aris2"] = Aris2
df_datos2 # confirmando realmente que se agregó la columna

CURSO  DEFINITIVA PROGRAMA  EDAD URBANO  PESO  ESTATURA  \
0    ESTADISTICA I         4.3     COMS    18     SI    55       164   
1    ESTADISTICA I         3.1     COMS    19     SI    65       165   
2    ESTADISTICA I         4.2     DERH    25     NO    56       170   
3    ESTADISTICA I         3.6     DERH    23     SI    65       170   
4    ESTADISTICA I         3.9     PSIC    19     SI    50       169   
..             ...         ...      ...   ...    ...   ...       ...   
116  ESTADISTICA I         3.5     PSIC    18     SI    60       163   
117  ESTADISTICA I         3.5     PSIC    18     SI    60       180   
118  ESTADISTICA I         2.5     COMS    21     NO    45       172   
119  ESTADISTICA I         3.0     COMS    20     NO    50       160   
120  ESTADISTICA I         3.5     DERH    21     NO    57       160   

     SATISFACC  SABER 11  PROMEDIO SEXO ESTRATO Herramientas  Aris2  
0            4       350      4.26    F      II         PTWF      0  
1            5       328      4.11    F       I         CYDT      1  
2            5       285      3.70    M       I         CYDT      2  
3            5       280      4.10    F      II         PCWF      3  
4            5       319      4.00    F      II         CYWF      4  
..         ...       ...       ...  ...     ...          ...    ...  
116          5       310      3.90    F     III         PTWF    116  
117          3       346      3.90    M      IV         PCWF    117  
118          5       221      3.70    F      II         CYWF    118  
119          4       332      3.50    F      II         CYDT    119  
120          4       293      3.50    F     III         PTWF    120  

[121 rows x 14 columns]

## **g. Ahora agregaremos a nuestra base de datos una columna de datos aleatorios**
- _randint_. La función _numpy.random.randint_ genera un array del tamaño indicado conteniendo números enteros aleatorios extraídos de una distribución discreta uniforme entre los intervalos dados (desde un valor inferior incluido, hasta un valor superior sin incluir):

In [ ]:
# Crear numeros enteros aleatorios entre 1 y 100
nota2 = np.random.randint(1,6, size = 121)
nota2

array([3, 2, 3, 4, 2, 5, 3, 4, 5, 1, 1, 2, 5, 5, 5, 1, 4, 2, 5, 5, 4, 2,
       4, 2, 1, 3, 2, 4, 4, 1, 1, 3, 4, 3, 5, 1, 1, 3, 5, 2, 5, 2, 2, 4,
       1, 2, 5, 3, 1, 2, 3, 1, 4, 4, 4, 4, 4, 4, 3, 1, 2, 4, 4, 1, 1, 4,
       1, 1, 5, 1, 3, 1, 1, 3, 5, 4, 1, 5, 4, 1, 4, 5, 1, 1, 3, 4, 1, 2,
       4, 4, 3, 5, 1, 4, 5, 5, 4, 5, 4, 5, 1, 1, 5, 2, 1, 3, 5, 4, 5, 4,
       3, 5, 5, 1, 3, 5, 5, 1, 1, 1, 5])

## **h. Ahora agregaremos esta nueva columna a nuestro DataFrame**

In [ ]:
## **h. Ahora agregaremos esta nueva columna a nuestro DataFrame**
df_datos2["nota2"] = nota2
df_datos2

CURSO  DEFINITIVA PROGRAMA  EDAD URBANO  PESO  ESTATURA  \
0    ESTADISTICA I         4.3     COMS    18     SI    55       164   
1    ESTADISTICA I         3.1     COMS    19     SI    65       165   
2    ESTADISTICA I         4.2     DERH    25     NO    56       170   
3    ESTADISTICA I         3.6     DERH    23     SI    65       170   
4    ESTADISTICA I         3.9     PSIC    19     SI    50       169   
..             ...         ...      ...   ...    ...   ...       ...   
116  ESTADISTICA I         3.5     PSIC    18     SI    60       163   
117  ESTADISTICA I         3.5     PSIC    18     SI    60       180   
118  ESTADISTICA I         2.5     COMS    21     NO    45       172   
119  ESTADISTICA I         3.0     COMS    20     NO    50       160   
120  ESTADISTICA I         3.5     DERH    21     NO    57       160   

     SATISFACC  SABER 11  PROMEDIO SEXO ESTRATO Herramientas  Aris2  nota2  
0            4       350      4.26    F      II         PTWF      0      3  
1            5       328      4.11    F       I         CYDT      1      2  
2            5       285      3.70    M       I         CYDT      2      3  
3            5       280      4.10    F      II         PCWF      3      4  
4            5       319      4.00    F      II         CYWF      4      2  
..         ...       ...       ...  ...     ...          ...    ...    ...  
116          5       310      3.90    F     III         PTWF    116      5  
117          3       346      3.90    M      IV         PCWF    117      1  
118          5       221      3.70    F      II         CYWF    118      1  
119          4       332      3.50    F      II         CYDT    119      1  
120          4       293      3.50    F     III         PTWF    120      5  

[121 rows x 15 columns]

#**44. Estadísticos y Descriptores númericos de un conjunto de Datos**

## **a. Descriptores numerícos de la Data**

In [ ]:
## **a. Descriptores numerícos de la Data**
df_datos2.describe()

DEFINITIVA        EDAD        PESO    ESTATURA  SATISFACC    SABER 11  \
count  121.000000  121.000000  121.000000  121.000000  121.00000  121.000000   
mean     3.773554   18.553719   63.239669  167.933884    4.53719  305.429752   
std      0.863980    1.653030   12.935755   12.369812    0.63300   43.059228   
min      1.100000   16.000000   37.000000  100.000000    3.00000  204.000000   
25%      3.500000   18.000000   55.000000  163.000000    4.00000  275.000000   
50%      4.000000   18.000000   61.000000  169.000000    5.00000  305.000000   
75%      4.500000   19.000000   73.000000  175.000000    5.00000  341.000000   
max      5.000000   27.000000   97.000000  192.000000    5.00000  380.000000   

         PROMEDIO       Aris2       nota2  
count  121.000000  121.000000  121.000000  
mean     4.030248   60.000000    3.049587  
std      0.332945   35.073732    1.537591  
min      3.200000    0.000000    1.000000  
25%      3.800000   30.000000    1.000000  
50%      4.000000   60.000000    3.000000  
75%      4.300000   90.000000    4.000000  
max      4.700000  120.000000    5.000000

## **b. Usando el método _astype()_ hacemos el cambio de tipo de variables para SATISFACC de numerica a categórica**

In [ ]:
# **b. Usando el método _astype()_ hacemos el cambio de tipo de variables para SATISFACC**
df_datos2["SATISFACC"]=df_datos2["SATISFACC"].astype('category')

## **c. Verificando el cambio en la variable hallamos nuevamente los descriptores numerícos de la Data**

In [ ]:
## **c. Verificando el cambio en la variable hallamos nuevamente los descriptores numerícos de la Data**
df_datos2.describe()

DEFINITIVA        EDAD        PESO    ESTATURA    SABER 11    PROMEDIO  \
count  121.000000  121.000000  121.000000  121.000000  121.000000  121.000000   
mean     3.773554   18.553719   63.239669  167.933884  305.429752    4.030248   
std      0.863980    1.653030   12.935755   12.369812   43.059228    0.332945   
min      1.100000   16.000000   37.000000  100.000000  204.000000    3.200000   
25%      3.500000   18.000000   55.000000  163.000000  275.000000    3.800000   
50%      4.000000   18.000000   61.000000  169.000000  305.000000    4.000000   
75%      4.500000   19.000000   73.000000  175.000000  341.000000    4.300000   
max      5.000000   27.000000   97.000000  192.000000  380.000000    4.700000   

            Aris2       nota2  
count  121.000000  121.000000  
mean    60.000000    3.049587  
std     35.073732    1.537591  
min      0.000000    1.000000  
25%     30.000000    1.000000  
50%     60.000000    3.000000  
75%     90.000000    4.000000  
max    120.000000    5.000000

## **d. Descriptores numerícos para una columna de la Data**


In [ ]:
## **d. Descriptores numerícos  de la Data**
df_datos2["EDAD"].describe()

count    121.000000
mean      18.553719
std        1.653030
min       16.000000
25%       18.000000
50%       18.000000
75%       19.000000
max       27.000000
Name: EDAD, dtype: float64

## **e. Editando a nuestros descriptores numéricos - La Media o Promedio**

In [ ]:
## **e. Editando a nuestros descriptores numéricos - La Media o Promedio**
print("La media de la variable EDAD es igual a", df_datos2["EDAD"].mean())

La media de la variable EDAD es igual a 18.553719008264462


## **f. Editando a nuestros descriptores numéricos - La Mediana**

In [ ]:
## **f. Editando a nuestros descriptores numéricos - La Mediana**
print("La mediana de la variable EDAD es igual a", df_datos2["EDAD"].median())

La mediana de la variable EDAD es igual a 18.0


## **g. Editando a nuestros descriptores numéricos - La Moda**

In [ ]:
## **g. Editando a nuestros descriptores numéricos - La Moda**
print("La moda de la variable EDAD es igual a", df_datos2["EDAD"].mode())

La moda de la variable EDAD es igual a 0    18
Name: EDAD, dtype: int64


## **h. Editando a nuestros descriptores numéricos - La Desvacion estandard**

In [ ]:
## **h. Editando a nuestros descriptores numéricos - La Desvacion estandard**
print("La desviación estandard de la variable EDAD es igual a", df_datos2["EDAD"].std())

La desviación estandard de la variable EDAD es igual a 1.6530296086435783


## **i. Editando a nuestros descriptores numéricos - Percentiles**

In [ ]:
## **i. Editando a nuestros descriptores numéricos - Percentiles**
print("El 75% de los estudiantes tienen edades por debajo de ", np.percentile(df_datos2["EDAD"], 75))

El 75% de los estudiantes tienen edades por debajo de  19.0


In [ ]:
## **i. Editando a nuestros descriptores numéricos - Percentiles**
print("El 85% de los estudiantes tienen edades por debajo de ", np.percentile(df_datos2["EDAD"], 85))

El 85% de los estudiantes tienen edades por debajo de  20.0


## **j. Editando a nuestros descriptores numéricos - Máximos y mínimos**

In [ ]:
## **j. Editando a nuestros descriptores numéricos - Máximos y mínimos**
print("Mis estudiantes tienen edades entre ",'(' ,df_datos2["EDAD"].min(),',' ,df_datos2["EDAD"].max(),')' )

Mis estudiantes tienen edades entre  ( 16 , 27 )


# **45. Vamos a Exportar nuestro Dataframes a limpios**

In [ ]:
# **45. Vamos a Exportar nuestro Dataframes a limpios**
df_datos2.to_csv('df_datos_LIMPIOS2.csv',index=False)